src/anet.c

```diff
@@ -420,11 +420,15 @@ static int anetTcpGenericConnect(char *err, const char *addr, int port,
         /* Try to create the socket and to connect it.
          * If we fail in the socket() call, or on connect(), we retry with
          * the next entry in servinfo. */
-        if ((s = socket(p->ai_family,p->ai_socktype,p->ai_protocol)) == -1)
+        #if defined(SOCK_NONBLOCK) && defined(SOCK_CLOEXEC)
+        if (flags & ANET_CONNECT_NONBLOCK) p->ai_socktype |= (SOCK_NONBLOCK | SOCK_CLOEXEC);
```

Comment:
`CLOEXEC` is something nice to have and it can tag along with `NONBLOCK` in one system call.

Do you prefer to make it optional in `anetCreateSocket`?

Reply:
i don't see why we need to bind them together.
if we consider it a nice-to-have, we can add it unconditionally (when we can) in anetCreateSocket.
and comment that it's a nice-to-have.
it doesn't need to depend on ANET_CONNECT_NONBLOCK

Reply:
Done.